lib/http/endpoint.js

```diff
@@ -238,6 +238,40 @@ const defaultEndpoint = endpointBase({
   errorWriter: defaultErrorWriter
 });
 
+const simpleResultWriter = (result, request, response, next) => {
+  // Expects a `result` with optional keys `body`, `status` and `headers`.
+  // No validation.
+  if (result.headers !== undefined) {
+    for (const [headername, headervalue] of Object.entries(result.headers)) {
+      response.setHeader(headername, headervalue);
+    }
+  }
+
+  if (result.status !== undefined) {
+    response.statusCode = result.status;
+  }
+
+  if (result.body !== undefined) {
+    if (result.body instanceof PartialPipe) {
+      result.body.streams.at(-1).on('error', streamErrorHandler(response));
+      result.body.with(response).pipeline((err) => next?.(err));
+    } else if (result.body.pipe != null) {
+      result.body.on('error', streamErrorHandler(response));
+      pipeline(result.body, response, (err) => err && next?.(err));
+    } else {
+      response.send(result.body);
+    }
+  } else {
+    // No body.
+    response.send();
+  }
+};
+
+const simpleEndpoint = endpointBase({
+  resultWriter: simpleResultWriter,
+  errorWriter: defaultErrorWriter,
+});
+
```

Comment:
I'd rather not have a new endpoint utility just to add 201/204 response codes and empty body (unless maybe it was well-tested and documented?). 

I think it would be simpler and more consistent with the rest of the code to settle for this basic "success" response that just returns `200 { success: true }`. Here's an example being used for deleting the other type of config we have: https://github.com/getodk/central-backend/blob/master/lib/resources/config.js#L52 For setting that other config, it looks like it returns what was sent to it, but in this case, i think it would be fine to return `200 { success: true }` for setting a preference, too.